include/seqan3/range/container/small_string.hpp

```diff
@@ -0,0 +1,265 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief A constexpr string implementation to manipulate string literals at compile time.
+ */
+
+#pragma once
+
+#include <seqan3/range/container/small_vector.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Implements a small string that can be used for compile time computations.
+ * \ingroup container
+ * \tparam N The size of the small string.
+ *
+ * This class provides a string type that can be constructed, evaluated and operated on at compile time.
+ * The stored string can be accessed as either a std::string or a c-style string through the respective member functions.
+ * All operations are constexpr except the conversion functions.
+ *
+ * Internally the string stores an array of size `N+1` containing the `0`-byte to allow conversion to
+ * c-style strings. This also means that when creating an instance of seqan3::small_string from a string literal
+ * containing a `0`-byte the size() of this small_string instance is one less than that of the literal.
```

Comment:
and instead of this (the second half is confusing because you are talking about size again, although it should be about capacity):


### Implementation notes

Internally the string stores a null-terminated array of size `capacity_ + 1` and the size of the string as a member. The smallest possible type is used for storage of the size so `small_string<30>` uses 32bytes of memory (one byte extra for the null-terminator and one byte to save the size).